packages/core/src/test/shared/env/resolveEnv.test.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import assert from 'assert'
+import * as resolveEnv from '../../../shared/env/resolveEnv'
+import sinon from 'sinon'
+import path from 'path'
+
+describe('resolveEnv', async function () {
+    let sandbox: sinon.SinonSandbox
+    beforeEach(function () {
+        sandbox = sinon.createSandbox()
+    })
+
+    afterEach(function () {
+        sandbox.restore()
+    })
+
+    // a copy of resolveEnv.mergeResolvedShellPath for stubbing
+    const testMergeResolveShellPath = async function mergeResolvedShellPath(
+        env: resolveEnv.IProcessEnvironment
+    ): Promise<typeof process.env> {
+        const shellEnv = await resolveEnv.getResolvedShellEnv(env)
+        // resolve failed or doesn't need to resolve
+        if (!shellEnv || Object.keys(shellEnv).length === 0) {
```

Comment:
another case that would be improved by https://github.com/aws/aws-toolkit-vscode/pull/5936/files#r1831813874